From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: larkwiot <22042193+larkwiot@users.noreply.github.com>
Date: Thu, 7 Oct 2021 11:10:20 -0400
Subject: [PATCH] 3481: Fix emulator's FilteredMemoryState write filter
 triggering after write

Ghidra should not suppress exceptions from scripts

Blanket-suppressing exceptions deprives writers of Ghidra Scripts the ability to handle or even see and know about exceptions in their MemoryAccessFilter's

Memory filters should get write event before write

Revert "Ghidra should not suppress exceptions from scripts"
---
 .../src/main/java/ghidra/app/emulator/FilteredMemoryState.java  | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Framework/Emulation/src/main/java/ghidra/app/emulator/FilteredMemoryState.java b/Ghidra/Framework/Emulation/src/main/java/ghidra/app/emulator/FilteredMemoryState.java
index cdbd538ca..95aa10ed3 100644
--- a/Ghidra/Framework/Emulation/src/main/java/ghidra/app/emulator/FilteredMemoryState.java
+++ b/Ghidra/Framework/Emulation/src/main/java/ghidra/app/emulator/FilteredMemoryState.java
@@ -46,7 +46,6 @@ class FilteredMemoryState extends DefaultMemoryState {
 
 	@Override
 	public void setChunk(byte[] res, AddressSpace spc, long off, int size) {
-		super.setChunk(res, spc, off, size);
 		if (filterEnabled && filter != null) {
 			filterEnabled = false;
 			try {
@@ -56,6 +55,7 @@ class FilteredMemoryState extends DefaultMemoryState {
 				filterEnabled = true;
 			}
 		}
+		super.setChunk(res, spc, off, size);
 	}
 
 	MemoryAccessFilter setFilter(MemoryAccessFilter filter) {
-- 
2.43.0

